Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/States/StateManager.cs

```diff
@@ -0,0 +1,422 @@
+        public InteractionState GetState(CoreInteractionState coreState)
+        {
+            return GetState(coreState.ToString());
+        }
+
+        /// <summary>
+        /// Gets and sets state given the state name and state value.
+        /// </summary>
+        /// <param name="stateName">The name of the state to set</param>
+        /// <param name="value">The new state value</param>
+        /// <returns>The state that was set</returns>
+        public InteractionState SetState(string stateName, int value)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                if (value > 0)
+                {
+                    SetStateOn(stateName);
+                }
+                else
+                {
+                    SetStateOff(stateName);
+                }
+            }
+            else
+            {
+                Debug.LogError($"The {stateName} state is not being tracked, add this state using AddState(state) to set it");
+            }
+
+            return state;
+        }
+
+        /// <summary>
+        /// Gets and sets a Core Interaction State given the value.
+        /// </summary>
+        /// <param name="coreState">The Core Interaction State to set</param>
+        /// <param name="value">The Core Interaction State's new value</param>
+        /// <returns>The Core Interaction State that was set</returns>
+        public InteractionState SetState(CoreInteractionState coreState, int value)
+        {
+            return SetState(coreState.ToString(), value);
+        }
+
+        /// <summary>
+        /// Gets and sets a state to On and invokes the OnStateActivated event. Setting a 
+        /// state on changes the state value to 1.
+        /// </summary>
+        /// <param name="stateName">The name of the state to set to on</param>
+        /// <returns>The state that was set to on</returns>
+        public InteractionState SetStateOn(string stateName)
+        {
+            InteractionState state = GetState(stateName);
+
+            if (state != null)
+            {
+                if (state.Value != 1)
+                {
+                    state.Value = 1;
+                    state.Active = 1;
+
+                    OnStateActivated.Invoke(state);
+
+                    // Only add the state to activeStates if it is not present 
+                    if (!activeStates.Contains(state))
+                    {
+                        activeStates.Add(state);
+                    }
+
+                    InteractionState defaultState = GetState(CoreInteractionState.Default);
+
+                    // If the state getting switched on is NOT the default state, then make sure the default state is off
+                    // The default state is only active when ALL other states are not active
+                    if (state.Name != "Default" && defaultState.Value == 1)
```

Comment:
Careful about magic strings and comparing to magic numbers. What is the difference between Active and Value?

Reply:
`Value` can take on values other than 0 and 1.  For example if we have a multi-dimensional toggle, the value could be greater than 1.  `Active` is a binary value and was changed to a bool instead of an int.  In the multi dimensional toggle example, if the value of the toggle is greater than 0, then it would be considered active.